### DISCUSSION TITLE
Error: Could not load model from file

### DISCUSSION THREAD

**Discussion by csamarchi** [eclipse-glsp/glsp#714] [General] [answered]
Hi,

I am running the GLSP vscode extension with node server and running into this issue: 

<img width="1048" alt="Screen Shot 2022-07-19 at 3 25 22 PM" src="https://user-images.githubusercontent.com/42501494/179841802-72ad105a-f260-41af-9bb7-4a4448a0c372.png">

Here is my `storage.ts` file:

<img width="521" alt="Screen Shot 2022-07-19 at 3 27 13 PM" src="https://user-images.githubusercontent.com/42501494/179842194-1f1fdda9-4edd-46ef-8278-1d780a689515.png">

It's saying it can't load from file but not saying why. The file path of the SourceUri is correct. How does the method `this.loadFromFile() `work? I cannot find docs anywhere. 

Also curious how `this.getSourceUri(action)` works? I know it is making a request with `RequestModelAction` to the client and receiving the action which is "requestModel" but where is it defining in the code that sourceUri is `'file:///Users/christine.samarchi/Desktop/FORCE/solution-builder-vscode-extension/graphical-editor/glsp-client-vscode/workspace/bad-bp.yaml` ?

Any help is much appreciated. Thank you

**Comment by csamarchi** [comment, marked as answer]
Solved this. The `loadFromFile` method does not work with our example because our data file is not JSON. Therefore, the method cannot read the file.

**Reply by tortmayr** [reply]
Yes that's correct. The `AbstractJsonModelStorage` can only be used for json data sources. For other data files you have to provide your own custom `SourceModelStorage`.
